pkg/doc/verifiable/common.go

```diff
@@ -59,6 +61,47 @@ func SingleKey(pubKey interface{}) PublicKeyFetcher {
 	}
 }
 
+// DIDKeyResolver resolves DID in order to find public keys for VC verification using vdri.Registry.
+// A source of DID could be issuer of VC or holder of VP. It can be also obtained from
+// JWS "issuer" claim or "verificationMethod" of Linked Data Proof.
+type DIDKeyResolver struct {
+	vdriRegistry vdri.Registry
+}
+
+// NewDIDKeyResolver creates DIDKeyResolver.
+func NewDIDKeyResolver(vdriRegistry vdri.Registry) *DIDKeyResolver {
+	return &DIDKeyResolver{vdriRegistry: vdriRegistry}
+}
+
+func (r *DIDKeyResolver) resolvePublicKey(issuerDID, keyID string) (interface{}, error) {
+	doc, err := r.vdriRegistry.Resolve(issuerDID)
+	if err != nil {
+		return nil, fmt.Errorf("resolve DID %s: %w", issuerDID, err)
+	}
+
+	if len(doc.PublicKey) == 0 {
```

Comment:
You probably don't need this check since this should be enforced during did creation. Anyway, the result would be not found error.

Reply:
Why do we default to first key?

Reply:
I see. Well, maybe we should return the first key if it's the only key and raise an error otherwise? @sandrask

Reply:
No - there are keys with different purposes in a DID document. We need to be able to select the appropriate key. @kdimak  @sandrask

Reply:
For now I would keep it simple with no defaults. User has to provide key id and you return not found if the key is not in the document. Later on we can add functions/implementations that will select appropriate type of key.

Reply:
OK, got it! I will change this soon.

Reply:
fixed